_posts/2017-09-18-minimap.md

```diff
@@ -0,0 +1,193 @@
+
+	public ContentRef<Material> CameraTexture
+	{
+		get { return cameraTexture; }
+		set { cameraTexture = value; }
+	}
+
+	public float MinimapSize
+	{
+		get { return minimapSize; }
+		set { minimapSize = value; }
+	}
+
+	public float BorderSize
+	{
+		get { return borderSize; }
+		set { borderSize = value; }
+	}
+
+	public float Padding
+	{
+		get { return padding; }
+		set { padding = value; }
+	}
+
+	private ContentRef<Material> cameraTexture = null;
+	private float minimapSize = 100;
+	private float borderSize = 5;
+	private float padding = 20;
+
+	public bool IsVisible(IDrawDevice device)
+	{
+		return (device.VisibilityMask & VisibilityFlag.AllGroups) != VisibilityFlag.None &&
+			   (device.VisibilityMask & VisibilityFlag.ScreenOverlay) != VisibilityFlag.None;
+	}
+
+	public void Draw(IDrawDevice device)
+	{
+		Canvas canvas = new Canvas(device);
+		Vector2 screenResolution = DualityApp.TargetResolution;
+
+		if (CameraTexture.IsAvailable)
+		{
+			float posX = screenResolution.X - minimapSize - borderSize - padding;
+			float posY = screenResolution.Y - minimapSize - borderSize - padding;
+			Vector2 minimapPos = new Vector2(posX, posY);
+
+			// draw the border
+			canvas.FillCircle(minimapPos.X, minimapPos.Y, minimapSize + borderSize);
+
+			// draw the minimap camera
+			canvas.State.SetMaterial(CameraTexture);
+			canvas.FillCircle(minimapPos.X, minimapPos.Y, minimapSize);
+		}
+	}
+}
+```
+
+> If you want to draw a square minimap, just replace the code above with `canvas.FillRect(...)`
+
+After that, add the `MinimapRenderer-Component` to your scene and asign the `Minimap-Texture` to the `CameraTexture-Property`.
+
+![Minimap Component](https://i.imgur.com/66ypSWD.png)
+
+Now the game should look like this:
+
+![Minimap](https://i.imgur.com/EiWl4Y1.png)
+
+
+Looks quite boring, right? Let's add some movement, so the camera can follow the player.
+Add both cameras, `MainCamera` and `MinimapCamera` to the `Player-GameObject` as children.
+
+We also create a new `PlayerController-Component` and add this to the `Player-GameObject`.
+
+```csharp
+[RequiredComponent(typeof(Transform))]
+public class PlayerController : Component, ICmpUpdatable
+{
+	private const float speed = 5f;
+
+	public void OnUpdate()
+	{
+		if (DualityApp.Keyboard.KeyPressed(Key.A))
+		{
+			Move(-Vector2.UnitX);
+		}
+
+		if (DualityApp.Keyboard.KeyPressed(Key.D))
+		{
+			Move(Vector2.UnitX);
+		}
+
+		if (DualityApp.Keyboard.KeyPressed(Key.W))
+		{
+			Move(-Vector2.UnitY);
+		}
+
+		if (DualityApp.Keyboard.KeyPressed(Key.S))
+		{
+			Move(Vector2.UnitY);
+		}
+	}
+
+	public void Move(Vector2 direction)
+	{
+		GameObj.Transform.MoveBy(direction * Time.TimeMult * speed);
+	}
+}
+```
+
+The scene view hierarchy should look like this:
+
+![Scene View](https://i.imgur.com/NYcG4Kl.png)
+
+... and the result:
+
+![](https://thumbs.gfycat.com/SoulfulRipeAdamsstaghornedbeetle-size_restricted.gif)
+
+I know, the controls sucks, but it should be fine for this tutorial.
```

Comment:
This line can be omitted imo. The controls are fine for the purpose of this tutorial.